### PR TITLE
Detection hits to legs

### PR DESCRIPTION
**Pull request by SecretAgent12**
hitbox code by @blooditor

### CODE REVIEW

File: src/main/java/com/flansmod/common/guns/raytracing/PlayerHitbox.java [hunk 1]
```diff
@@ -176,8 +176,10 @@ public float hitByBullet(EntityBullet bullet, float penetratingPower) {
 
         if (type == EnumHitboxType.HEAD) {
             totalPenetrationResistance = headPenRes;
+        } else if(type == EnumHitboxType.LEGS){
+            totalPenetrationResistance = legsPenRes + feetPenRes;
```
Comment (GoldSloth) [inline src/main/java/com/flansmod/common/guns/raytracing/PlayerHitbox.java:180]: change to defaults for legPenRes and feetPenRes?

File: src/main/java/com/flansmod/common/guns/raytracing/PlayerHitbox.java [hunk 2]
```diff
@@ -192,9 +194,12 @@ public float hitByBullet(EntityBullet bullet, float penetratingPower) {
         if (type == EnumHitboxType.HEAD) {
             damageModifier *= 2F;
             bullet.lastHitHeadshot = true;
+        } else if(type == EnumHitboxType.LEGS) {
+            damageModifier *=0.6F;
         }
-
+        System.out.println(damageModifier);
```
Comment (GoldSloth) [inline src/main/java/com/flansmod/common/guns/raytracing/PlayerHitbox.java]: println, please remove

File: src/main/java/com/flansmod/common/guns/raytracing/PlayerHitbox.java [hunk 3]
```diff
@@ -203,7 +208,7 @@ public float hitByBullet(EntityBullet bullet, float penetratingPower) {
                 float hitDamage = bullet.damage * bullet.type.damageVsPlayer * damageModifier;
                 //Create a damage source object
                 DamageSource damagesource = bullet.owner == null ? DamageSource.generic : bullet.getBulletDamage(type == EnumHitboxType.HEAD);
-
+                System.out.println(hitDamage);
```
Comment (GoldSloth) [inline src/main/java/com/flansmod/common/guns/raytracing/PlayerHitbox.java]: please remove

File: src/main/java/com/flansmod/common/guns/raytracing/PlayerHitbox.java (hunk 1 above)
Comment (SecretAgent12) [reply src/main/java/com/flansmod/common/guns/raytracing/PlayerHitbox.java:180]: yes

Comment (SecretAgent12) [reply src/main/java/com/flansmod/common/guns/raytracing/PlayerHitbox.java:180]: how?

Comment (GoldSloth) [reply src/main/java/com/flansmod/common/guns/raytracing/PlayerHitbox.java:180]: See line 166, it's a ternary statement to use the armour penRes value OR the default pen res value.

So either this part reverts back to
legsOrBodyOrFeet = legs+feet+chest
head=head

Or the defaults will be changed so:
legs+feet=1
chest=1
head=1
And people's pen res amounts for armour will work differently.

**Comment by GoldSloth** [comment]
Looks good, thank you sire.

**Pull request merged**